### ISSUE TITLE
No re-rendering after relative context update

### ISSUE THREAD

**Issue by smoldaner** [ractivejs/ractive#3049]
## Description:

Using the example from the [current context documentation](https://ractive.js.org/api/#current-context):

If the ``spoon2`` value is set by an observer, it's not readable/updated by a relative keypath (inside the matrix). 

Expected:

    Outside the matrix, you have SPOON
    Inside the matrix, you think you have SPOON.
    In reality, there is no

If I add ``spoon2`` to ``data`` with an initial value (remove comment from code), everything works as expected.

## Versions affected:

edge (0.9.3)

## Platforms affected:

Chrome 60.0.3112.78, others as well

## Reproduction:

s. this [fiddle](https://jsfiddle.net/03kgrohd/2/)

### Template

    <script src="https://unpkg.com/ractive"></script>
    <script type="text/html" id="template">
	<div>Outside the matrix, you have {{ spoon2 }}</div>
    {{# matrix }}
      <div>Inside the matrix, you think you have {{ spoon2 }}.</div>
      <div>In reality, there is no {{ ./spoon2 }}</div>
    {{/}}
    </script>

### Code

    const r = window.r = new Ractive({
	el: '#main',
    template: '#template',
    data: {
      //spoon2: 'SPOON2'
      spoon: 'SPOON',
      matrix: {
          agent: 'Smith'
         // There is no spoon
      }
     },
    oninit: function() {
  	var component = this
  	this.observe('spoon', function(newVal){
    	component.set('spoon2', newVal)
        })
      }
    });

**Comment by evs-chris** [comment]
TL;DR if the data doesn't exist, ambiguous references resolve to the immediate context.

This is actually working as intended. When you reference data that doesn't yet exist _somewhere_ up the context hierarchy, it automatically resolves in the nearest context - meaning that the first `spoon2` resolves to `~/spoon2`, the second just inside the `matrix` resolves to `matrix.spoon2` (because `~/spoon2` doesn't actually exist), and the third is not ambiguous, so it resolves to `matrix.spoon2`. Having `spoon2` provided in the data means that `~/spoon2` _does_ exist, so the ambiguous reference inside `matrix` can resolve.

You can either leave data uninitialized and use unambiguous references, or you can initialize it and use ambiguous references. Unambiguous references are always preferable, because they _always_ resolve faster and with fewer surprises in situations like this, particularly when a conditional is rendered, data appears that would affect resolutions within the conditional, and the conditional is unrendered and rerendered (sometimes much) later. Then you get different results in the conditional with no obvious link between them that would have caused the differences.

**Comment by smoldaner** [comment]
So this is a breaking change in version 0.9? With version 0.8.x the code worked as expected. 

Thanks for clarification!

**Comment by evs-chris** [comment]
It is indeed: https://ractive.js.org/support/#unresolved-references

The unresolved process had some subtle bugs and caused inconsistencies in render in certain circumstances.
